perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java

```diff
@@ -3643,6 +3643,16 @@ private int processApplication(PerunSession session, Application application, Li
 
 	}
 
+	@Override
+	public void updateApplicationType(PerunSession session, Application application) {
+
+		// TODO - add authorization (and add to rpc)
+
+		if (jdbc.update("update application set apptype=? where id=?", application.getType(), application.getId()) > 0) {
```

Comment:
Must be ``application.getType().toString()`` otherwise jdbc driver cannot guess the type: ```Can't infer the SQL type to use for an instance of cz.metacentrum.perun.registrar.model.Application$AppType.```

Reply:
Ok, fixed.